public/scripts/editor/js/fc/bramble-input-field.js

```diff
@@ -0,0 +1,139 @@
+// This creates an input field that can morph into a static placeholder
+// when a user wants to save their input.
+define(function(require) {
+  var $ = require("jquery");
+
+  function triggerListeners(context, listenerList, args) {
+    if(!listenerList) {
+      return;
+    }
+
+    listenerList.forEach(function(listener) {
+      listener.apply(context, args);
+    });
+  }
+
+  function create(context) {
+    var element;
+
+    if(context.mode === InputField.Modes.STATIC) {
+      element = $("<span></span>");
+    } else {
+      element = $("<input type=\"text\" />");
+    }
+
+    element.attr("id", context.id);
+    element.addClass(context._element.attr("class"));
+
+    return element;
+  }
+
+  function generateElement(mode) {
+    return mode === InputField.Modes.STATIC ?
+           $("<span></span>") : $("<input type=\"text\" />");
+  }
+
+  // This creates a morphable input field and takes in the
+  // container (should be a div as a DOM element or JQuery object)
+  // that will contain the input field. The constructor also takes
+  // a second optional boolean argument that will force the input field
+  // to be added before any other elements in the container (by default,
+  // it is appended to the container)
+  // The input field starts of as a static placeholder
+  function InputField(container, prepend) {
+    // Internal values
+    this._container = container instanceof $ ? container : $(container);
+    this._element = generateElement(InputField.Modes.STATIC);
```

Comment:
Ah, okay. I tend to only factor out logic if the concerns change or if there's redundancy.